server/imports/core/connection/index.js

```diff
@@ -170,92 +277,134 @@ Connection.prototype = {
+    Database.read({ type: Database.types.Connections }).forEach(
+      (oldConnection) => {
+        // if there's a name (was mandatory) property it's old.
+        if (oldConnection.name) {
+          let connection = { options: {} };
+          if (oldConnection.url) {
+            connection = this.parseUrl({ url: oldConnection.url });
+            connection.url = oldConnection.url;
+          }
+
+          connection._id = oldConnection._id;
+          connection.connectionName = oldConnection.name;
+
+          migrateSSHPart(oldConnection, connection);
+
+          if (oldConnection.host && oldConnection.port) {
+            connection.servers = [
+              {
+                host: oldConnection.host,
+                port: oldConnection.port,
+              },
+            ];
+          }
+          if (oldConnection.readFromSecondary) {
+            connection.options.readPreference = 'secondary';
+          } else connection.options.readPreference = 'primary';
+
+          if (oldConnection.databaseName) {
+            connection.databaseName = oldConnection.databaseName;
+          }
+          if (oldConnection.user && oldConnection.password) {
+            connection.scram_sha_1 = {
+              username: oldConnection.user,
+              password: oldConnection.password,
+            };
+            if (oldConnection.authDatabaseName) {
+              connection.scram_sha_1.authSource =
+                oldConnection.authDatabaseName;
+            }
+            connection.authenticationType = 'scram_sha_1';
+          }
+
+          if (oldConnection.useSsl || oldConnection.sslCertificatePath) {
+            connection.ssl = { enabled: true };
+          }
+
+          if (oldConnection.x509Username) {
+            connection.authenticationType = 'mongodb_x509';
+            connection.mongodb_x509 = { username: oldConnection.x509Username };
+            delete connection.ssl;
+          }
+
+          migrateSSLPart(oldConnection, connection);
+
+          connectionsAfterMigration.push(connection);
         }
-
-        if (oldConnection.useSsl || oldConnection.sslCertificatePath) connection.ssl = { enabled: true };
-
-        if (oldConnection.x509Username) {
-          connection.authenticationType = 'mongodb_x509';
-          connection.mongodb_x509 = { username: oldConnection.x509Username };
-          delete connection.ssl;
-        }
-
-        migrateSSLPart(oldConnection, connection);
-
-        connectionsAfterMigration.push(connection);
       }
-    });
-
+    );
 
     Database.remove({ type: Database.types.Connections });
-    connectionsAfterMigration.forEach(conn => Database.create({ type: Database.types.Connections, document: conn }));
-    Database.update({ type: Database.types.Settings, selector: {}, modifier: { $set: { isMigrationDone: true } } });
+    connectionsAfterMigration.forEach((conn) =>
+      Database.create({ type: Database.types.Connections, document: conn })
```

Comment:
Codacy has a fix for the issue: [Expected no linebreak before this expression.](https://app.codacy.com/gh/nosqlclient/nosqlclient/pullRequest?prid=7730964)
```suggestion
    );
```